interfaces/builtin/desktop.go

```diff
@@ -38,13 +38,27 @@ const desktopBaseDeclarationSlots = `
   desktop:
     allow-installation:
       slot-snap-type:
+        - app
```

Comment:
we might want to keep this as is and allow single snaps one by one to have this slot?

Reply:
If the base declaration only allows `slot-snap-type: [core]`, then it is impossible to install an unpublished snap with the slot. It doesn't hit the "default allow" path for this case  like other declaration constraints. It'd be a pain to only be able to test a snap after it has been published and gained a snap-declaration from the store.

Reply:
we have hit similar issues before and we have other use cases. Basically there are slot for which we want to be careful about the apps that have them but OTOH we don't want to stop development of such snaps. The simplest if awkward approach is to change the checks we do for locally installed snaps to consider only the allow-installation rules. Then the strange:

```
allow-installation:
   slot-snap-type:
   - core
   - app
 deny-installation:
   slot-snap-type:
      - app
```

Would achieve what we intend.

A different approach would be to introduce something like "dangerous-app" that is a subset of app but it is awkward in different ways.

Reply:
The allow-installation/deny-installation pair seems to work with current snapd. I've also added policy unit tests to verify, and the spread test using an unpublished snap with a desktop slot is working.